gensim/test/test_fasttext.py

```diff
@@ -1300,6 +1306,253 @@ def _run(self, fin):
+        self._check_roundtrip_model_model(model_params)
+
+    def test_round_trip_model_model_cbow(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 0,
+            "negative": 5,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_model_model(model_params)
+
+
+def _read_binary_file(fname):
+    with open(fname, "rb") as f:
+        data = f.read()
+    return data
+
+
+class SaveFacebookFormatFileGensimTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create binary fastText file model1.bin using gensim
+    + load file model1.bin to variable `model`
+    + save `model` to model2.bin
+    + check if files model1.bin and model2.bin are byte identical
+    """
+
+    def _check_roundtrip_file_file(self, model_params):
+        with temporary_file("roundtrip_file_to_file1.bin") as fpath1, \
+            temporary_file("roundtrip_file_to_file2.bin") as fpath2:
+            _create_and_save_test_model(fpath1, model_params)
+            model = gensim.models.fasttext.load_facebook_model(fpath1)
+            gensim.models.fasttext.save_facebook_model(model, fpath2)
+            self.assertEqual(_read_binary_file(fpath1), _read_binary_file(fpath2))
+
+    def test_roundtrip_file_file_skipgram(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 1,
+            "negative": 0,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_file_file(model_params)
+
+    def test_roundtrip_file_file_cbow(self):
+        model_params = {
+            "size": 10,
+            "min_count": 1,
+            "hs": 1,
+            "sg": 0,
+            "negative": 0,
+            "seed": 42,
+            "workers": 1}
+        self._check_roundtrip_file_file(model_params)
+
+
+def _save_test_model(out_base_fname, model_params, fasttext_cmd):
+    inp_fname = datapath('lee_background.cor')
+
+    model_type = "cbow" if model_params["sg"] == 0 else "skipgram"
+    size = str(model_params["size"])
+    seed = str(model_params["seed"])
+
+    cmd = fasttext_cmd + " " + model_type + " -input " + inp_fname + \
+            " -output " + out_base_fname + " -dim " + size + " -seed " + seed
+    subprocess.run(cmd, shell=True)
+
+
+class SaveFacebookFormatFileFastTextTest(unittest.TestCase):
+    """
+    This class containts tests that check the following scenario:
+
+    + create binary fastText file model1.bin using facebook_binary (FT)
+    + load file model1.bin to variable `model`
+    + save `model` to model2.bin using gensim
+    + check if files model1.bin and model2.bin are byte-identical
+
+    Requires env. variable FT_HOME to point to location of Facebook fastText binary
+    """
+
+    def _check_roundtrip_file_file(self, model_params):
+        ft_home = os.environ.get("FT_HOME", None)
```

Comment:
You're doing this check in three separate places (here, and in each of the two tests). Move it to module scope:

```python
FT_HOME = os.environ.get("FT_HOME")
FT_CMD = ...
```

and use the result in its stead.

You can also skip the entire suite if FT_HOME is missing, instead of skipping individual tests.

Also, you don't need to pass None to `get()`, as that is the default behavior.

Reply:
This is a nice one. A lot of simplification.

Corrected.